src/compiler/parser.ts

```diff
@@ -4375,9 +4515,24 @@ module ts {
                 //      [~GeneratorParameter]ClassHeritage[?Yield]opt { ClassBody[?Yield]opt }
                 //      [+GeneratorParameter] ClassHeritageopt { ClassBodyopt }
 
-                node.members = inGeneratorParameterContext()
-                    ? doOutsideOfYieldContext(parseClassMembers)
-                    : parseClassMembers();
+                if (inGeneratorParameterContext()) {

```

Comment:
this pattern has repeated several times.  Is is possible to have a helper for this?